### PR TITLE
Custom errors new option

### LINKED ISSUE #32: Custom errors
Errors should have a custom form and not a `KBXXX: message`

Proposal:

```javascript
const DEVPORTAL_BASE_URL = 'https://developers.rsk.co/rif/rns/libs/javascript/Errors/'

class RNSError extends Error {
  constructor(id) {
    const { message, name } = errorMessages(id);
    super(message);
    this.name = name;
    this.ref = `${DEVPORTAL_BASE_URL}#${id}`;
  }
}
```

This will also require to implement ids in RSK DevPortal

### PR DESCRIPTION
**Pull request by javiesses**

### CODE REVIEW

**Comment by javiesses** [comment]
Closes #32

File: test/utils.ts [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
   let error;
   try {
     fn();
   } catch (_error) {
-    error = _error.message;
+    error = _error;
   } finally {
-    expect(error).toBe(expectedError);
+    const expectedErrorId = (errors as ErrorDictionary)[expectedError].id;
+    expect(error.id).toBe(expectedErrorId);
```
Comment (ilanolkies) [inline test/utils.ts]:
```suggestion
    expect(error).toEqual(new RNSError(expectedError));
```

??

File: package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "test": "npm run test:local && npm run test:public && npm run lint",
     "test:local": "npx jest --coverage test/local/",
     "test:public": "npx jest --coverage test/public-nodes/",
-    "lint": "npx eslint 'src/**' 'test/**'",
+    "lint": "npx eslint 'src/**.ts' 'src/**/**.ts' 'test/**'",
```
Comment (ilanolkies) [inline package.json:28]:
```suggestion
    "lint": "npm run lint:src && npm run lint:test",
    "lint:src": "npx eslint ./src",
    "lint:test": "npx eslint ./test",
```

**Review by ilanolkies — CHANGES_REQUESTED** [review]
Great job!

File: package.json (hunk 2 above)
Comment (ilanolkies) [reply package.json:28]: Paths in eslint are strange...

**Review by ilanolkies — APPROVED** [review]

**Pull request merged**